rootfs/scheduler/__init__.py

```diff
@@ -768,27 +768,15 @@ def _delete_namespace(self, namespace):
     # REPLICATION CONTROLLER #
 
     def _get_old_rc(self, namespace, app_type):
-        url = self._api("/namespaces/{}/replicationcontrollers", namespace)
-        resp = self.session.get(url)
-        if unhealthy(resp.status_code):
-            error(resp, 'get ReplicationControllers in Namespace "{}"', namespace)
-
-        exists = False
-        prev_rc = []
-        for rc in resp.json()['items']:
-            if (
-                'app' in rc['spec']['selector'] and
-                namespace == rc['metadata']['labels']['app'] and
-                'type' in rc['spec']['selector'] and
-                app_type == rc['spec']['selector']['type']
-            ):
-                exists = True
-                prev_rc = rc
-                break
-        if exists:
-            return prev_rc
+        labels = {
+            'app': namespace,
+            'type': app_type
+        }
+        controllers = self._get_rcs(namespace, labels=labels).json()
+        if len(controllers['items']) == 0:
+            return False
```

Comment:
pretty much. Unless we're specifically checking for `None` I don't think there's much of a difference.